pkg/controller/snapshotter/snapshotter.go

```diff
@@ -39,14 +39,14 @@ const (
 	defaultExponentialBackOffOnError = true
 )
 
-// VolumeSnapshotter does the "heavy lifting": it spawns gouroutines that talk to the
+// VolumeSnapshotter does the "heavy lifting": it spawns goroutines that talk to the
 // backend to actually perform the operations on the storage devices.
 // It creates and deletes the snapshots and promotes snapshots to volumes (PV). The create
 // and delete operations need to be idempotent and count with the fact the API object writes
 type VolumeSnapshotter interface {
-	CreateVolumeSnapshot(snapshotName string, snapshotSpec *tprv1.VolumeSnapshotSpec)
-	DeleteVolumeSnapshot(snapshotName string, snapshotSpec *tprv1.VolumeSnapshotSpec)
-	PromoteVolumeSnapshotToPV(snapshotName string, snapshotSpec *tprv1.VolumeSnapshotSpec)
+	CreateVolumeSnapshot(snapshotName string, snapshot *tprv1.VolumeSnapshot)
+	DeleteVolumeSnapshot(snapshotName string, snapshotc *tprv1.VolumeSnapshot)
```

Comment:
Small typo: "snapshotc". Moreover... Do we still need to pass the snaphsotName to all these functions? We have the whole VolumeSnapshot...